web/views.py

```
@@ -6929,3 +6931,165 @@
+            logger.error(f"Error saving fact: {str(e)}")
+            # Still return the generated fact even if we couldn't save it
+
+        return JsonResponse({"fact": fact})
+    except Subject.DoesNotExist:
+        return JsonResponse({"error": "Subject not found"}, status=404)
+    except Exception as e:
+        print(f"Error generating fact with Gemini: {str(e)}")
+        return JsonResponse({"fact": get_fallback_fact(subject.name)})
+
+
+def generate_fact_for_subject(subject_name):
```

Ruff (0.8.2)

Missing return type annotation for public function `generate_fact_for_subject`

(ANN201)